osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/ExplodePiece.cs

```diff
@@ -18,13 +25,36 @@ public ExplodePiece()
 
             Blending = BlendingParameters.Additive;
             Alpha = 0;
+        }
 
-            Child = new TrianglesPiece
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Child = triangles = new TrianglesPiece
             {
                 Blending = BlendingParameters.Additive,
                 RelativeSizeAxes = Axes.Both,
                 Alpha = 0.2f,
             };
+
+            drawableObject.HitObjectApplied += onHitObjectApplied;
+            onHitObjectApplied(drawableObject);
+        }
+
+        private void onHitObjectApplied(DrawableHitObject obj)
+        {
+            if (obj.HitObject == null)
+                return;
```

Comment:
Are these null-checks really required? From what I can tell DHO cannot invoke `HitObjectApplied` with a null `HitObject`, or else they'd throw here in the first place:

https://github.com/ppy/osu/blob/851c7d524fe6f3ce6b638718ec4abf454c58d2df/osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs#L191

Or is this for safety in case that the object is freed in the meantime? Can't imagine how that'd happen, though...

Reply:
Shouldn't be required - not sure why I left this here.

Reply:
Ah no - I added it for safety because of the immediate invocation in `load()`, at which point DHOs can't be assumed to have a hitobject (although they might in tests, which is why that invocation exists in the first place).

Reply:
Could this potentially work?

![image](https://user-images.githubusercontent.com/191335/99511400-8d42d780-29cb-11eb-8948-4779761675eb.png)

Reply:
No because of the second care I mentioned in my edit - if the pool is empty, the new drawable is loaded after `Apply()`.